etcdserver/api/v2http/client_auth.go

```diff
@@ -90,8 +90,7 @@ func hasKeyPrefixAccess(sec auth.Store, r *http.Request, key string, recursive b
 		plog.Warningf("auth: no such user: %s.", username)
 		return false
 	}
-	authAsUser := sec.CheckPassword(user, password)
-	if !authAsUser {
```

Comment:
I want to know why tom:tom does not work any more. I know this can solve the issue `tom:` does work.